pep-9999.rst

```diff
@@ -0,0 +1,437 @@
+will only alert us to update call sites that deal with the base type directly.
+But the type checker can only see the new code, not the change we made, so it
+has no way of knowing that we probably also needed to rename the same method on
+child classes.
+
+A type checker will happily accept this code, even though we are likely
+introducing bugs:
+
+.. code-block:: python
+
+    class Parent:
+        # Rename this method
+        def new_foo(self, x: int) -> int:
+            return x
+
+    class Child(Parent):
+        # This (unchanged) method used to override `foo` but is unrelated to `new_foo`
+        def foo(self, x: int) -> int:
+            return x + 1
+
+    def parent_callsite(parent: Parent) -> None:
+        # If we pass a Child instance we’ll now run Parent.new_foo - likely a bug
+        parent.new_foo(1)
+
+    def child_callsite(child: Child) -> None:
+        # We probably wanted to invoke new_foo here. Instead, we forked the method
+        child.foo(1)
+
+
+This code will type check, but there are two potential sources of bugs:
+
+- If we pass a ``Child`` instance to the parent_callsite function, it will
+  invoke the implementation in ``Parent.new_foo``. rather than ``Child.foo``.
+  This is probably a bug - we presumably would not have written ``Child.foo`` in
+  the first place if we didn’t need custom behavior.
+- Our system was likely relying on ``Child.foo``  behaving in a similar way to
+  ``Parent.foo``. But unless we catch this early, we have now forked the
+  methods, and future refactors it is likely no one will realize that major
+  changes to the behavior of new_foo likely require updating ``Child.foo`` as
+  well, which could lead to major bugs later.
+
+The incorrectly-refactored code is type-safe, but is probably not what we
+intended and could cause our system to behave incorrectly. The bug can be
+difficult to track down because our new code likely does execute without
+throwing exceptions. Tests are less likely to catch the problem, and silent
+errors can take longer to track down in production.
+
+We are aware of several production outages in multiple typed codebases caused by
+such incorrect refactors. This is our primary motivation for adding an ``@override``
+decorator to the type system, which lets developers express the relationship
+between ``Parent.foo`` and ``Child.foo`` so that type checkers can detect the problem.
+
+
+Subclass Implementations Become More Explicit
+---------------------------------------------
+
+We believe that explicit overrides will make unfamiliar code easier to read than
+implicit overrides. A developer reading the implementation of a subclass that
+uses ``@override``  can immediately see which methods are overriding
+functionality in some base class; without this decorator, the only way to
+quickly find out is using a static analysis tool.
+
+
+Modern Integrated Development Environments (IDEs) often provide the ability to
```

Comment:
This doesn't feel like it's covered by the subsection title. It may work better in a "Rejected ideas" section.

Reply:
Whoops, I missed a subsection header when converting the google doc to rst.

I'm happy to move this section or just take it out, but first I'll add the header back in and we can see what it looks like then.